social_navigation_layers/src/human_layer.cpp

```diff
@@ -0,0 +1,223 @@
+    return;
+
+  boost::recursive_mutex::scoped_lock lock(lock_);
+  if (!enabled_)
+    return;
+
+  std::list<people_msgs::Person>::iterator p_it;
+  costmap_2d::Costmap2D* costmap = layered_costmap_->getCostmap();
+  double res = costmap->getResolution();
+  double base = get_radius(inscribed_radius_, cutoff_, variance_);
+  // Width of square bounding box for a person in units of costmap cells
+  unsigned int width = std::max(1, static_cast<int>((2.0 * base) / res));
+
+  for (p_it = transformed_people_.begin(); p_it != transformed_people_.end(); ++p_it)
+  {
+    people_msgs::Person person = *p_it;
+
+    // Global frame coordinates of person (meters)
+    double cx = person.position.x;
+    double cy = person.position.y;
+
+    // Offset coordinates of person (meters). For a typical map, this is
+    // the lower-left corner of the bounding box for the costmap cells
+    // we are going to modify.
+    double ox = cx - base;
+    double oy = cy - base;
+
+    // Convert offset coords to discretized coordinates in units of
+    // costmap cells
+    int dx, dy;
+    costmap->worldToMapNoBounds(ox, oy, dx, dy);
+
+    // Calculate size of bounding box in units of costmap cells
+    int start_x = 0, start_y = 0, end_x = width, end_y = width;
+    if (dx < 0)
+      start_x = -dx;
+    else if (dx + width > costmap->getSizeInCellsX())
+      end_x = std::max(0, static_cast<int>(costmap->getSizeInCellsX()) - dx);
+
+    if (static_cast<int>(start_x + dx) < min_i)
+      start_x = min_i - dx;
+    if (static_cast<int>(end_x + dx) > max_i)
+      end_x = max_i - dx;
+
+    if (dy < 0)
+      start_y = -dy;
+    else if (dy + width > costmap->getSizeInCellsY())
+      end_y = std::max(0, static_cast<int>(costmap->getSizeInCellsY()) - dy);
+
+    if (static_cast<int>(start_y + dy) < min_j)
+      start_y = min_j - dy;
+    if (static_cast<int>(end_y + dy) > max_j)
+      end_y = max_j - dy;
+
+    // Shift bounding box start by half of a costmap cell to represent
+    // the center of the cell, instead of the lower-left corner, to
+    // calculate the cost. (bx, by) are in units of meters.
+    double bx = ox + res / 2;
+    double by = oy + res / 2;
+    // Calculate the cost of each cell in the bounding box around person
+    for (int i = start_x; i < end_x; i++)
+    {
+      for (int j = start_y; j < end_y; j++)
+      {
+        unsigned char old_cost = costmap->getCost(i + dx, j + dy);
+        if (old_cost == costmap_2d::NO_INFORMATION)
+          // Do not change costmap if there is no information
+          // previously. Most commonly applies past the edge of the map.
+          continue;
+
+        double x = bx + i * res;
+        double y = by + j * res;
+        double a = calculate_cost(x, y, cx, cy, variance_, lethal_radius_, inscribed_radius_);
+
+        if (a < cutoff_)
+          // Don't bother updating the costmap for costs that are small
+          continue;
+
+        unsigned char cvalue = (unsigned char) a;
```

Comment:
please make this a `static_cast<unsigned char>`

Reply:
Done.